task/buildah/0.1/buildah.yaml

```
@@ -267,7 +267,12 @@ spec:
         cp -r "$(workspaces.source.path)/cachi2" /tmp/
         chmod -R go+rwX /tmp/cachi2
         VOLUME_MOUNTS="--volume /tmp/cachi2:/cachi2"
-        sed -i 's|^\s*run |RUN . /cachi2/cachi2.env \&\& \\\n    |i' "$dockerfile_path"
+        # Read in the whole file (https://unix.stackexchange.com/questions/533277), then
+        # for each RUN ... line insert the cachi2.env command *after* any options like --mount
+        sed -E -i \
+            -e 'H;1h;$!d;x' \
+            -e 's@^\s*(run((\s|\\\n)+-\S+)*(\s|\\\n)+)@\1. /cachi2/cachi2.env \&\& \\\n    @igM'
+            "$dockerfile_path"
```

Finally found the error message in the e2e-test output this time:
```
[e2e-tests : e2e-test]   sed: no input files
```
```suggestion
            -e 's@^\s*(run((\s|\\\n)+-\S+)*(\s|\\\n)+)@\1. /cachi2/cachi2.env \&\& \\\n    @igM' \
            "$dockerfile_path"
```

---

Bah - obviously I never tested the tweaked version in real life, just that fragment in isolation :-(  -  thank goodness for the e2e tests! (I don't have log access, so I couldn't see how they were failing.)  I've now pushed a fixed up version, also adding the changes for 0.2/ as well.